automated-comments/csharp/general/do_not_use_nested_if_statement.md

```diff
@@ -0,0 +1,35 @@
+You are currently using nested if-statements. In general, it is usually best to avoid deep nesting, as it makes for harder reading. Consider the following two pieces of code:
```

Comment:
@mikedamay said:

"I have never understood the preference for early returns or the obsession with cyclomatic complexity although I concede I am in the minority. I was slightly heartened to see that the Stack Exchange Software Engineering question https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement was closed as opinion based so there is hope for my tribe."

Reply:
Hmmm, let's ask some other people's opinions then. @robkeim @SleeplessByte what do you think about this comment and the guideline itself?

Reply:
Sounds good - as I said in my original reply on Slack - there are no show stoppers.  I would certainly be comfortable with your original wording.

Reply:
Nested ifs in generals aren't bad, but I do care about cognitive complexity (and therefor also about cyclomatic complexity). Nested if statements often _hide_ other smells or issues. I think it's fine to point this out.

I would suggest:

```suggestion
You are currently using nested if-statements. When possible, try to avoid deep nesting, as it can lead to code that is harder to follow. Consider the following two pieces of code:
```

Reply:
Great suggestion @SleeplessByte !

Reply:
I think it's hard to find a strict black and white rule that applies in this case, but I also think it should be pointed out as a potential area of improvement.